pkg/controller/controller.go

```diff
@@ -250,12 +260,45 @@ func (c *Controller) Run(stop chan struct{}) {
 		log.Panicln("Timed out waiting for namespace cache to sync.")
 	}
 
+	defer c.queue.ShutDown()
+	go wait.Until(c.runWorker, 0, stop)
+
+	// TODO, change once athenz domain cr watch is in place, add individual
+	// items to queue
 	for {
-		err := c.sync()
-		if err != nil {
-			log.Println("Error running sync:", err)
-		}
+		c.queue.Add(queueKey)
 		log.Println("Sleeping for", c.pollInterval)
 		time.Sleep(c.pollInterval)
 	}
 }
+
+// runWorker calls processNextItem to process events of the work queue
+func (c *Controller) runWorker() {
+	for c.processNextItem() {
+	}
+}
+
+// processNextItem takes an item off the queue and calls the controllers sync
+// function, handles the logic of requeuing in case any errors occur
+func (c *Controller) processNextItem() bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer c.queue.Done(key)
+
+	// TODO, change to new sync function
+	err := c.sync()
+	if err != nil {
+		log.Printf("Error syncing athenz state for key %s: %s", key, err)
```

Comment:
might need a `err.Error()` for `%s` ?

Reply:
@prabushyam This is not needed, will get automatically converted to string.